website/addons/base/__init__.py

```diff
@@ -350,6 +351,46 @@ def deep_url(self):
     def revision(self):
         return getattr(self, '_revision', None)
 
+    def add_tag(self, tag, auth, node, file_name, save=True):
```

Comment:
Needs a docstring.

Be sure to indicate if `tag` must be a string, or if a `Tag` object might be used as well.